src/data_structures/fenwick_tree.js

```diff
@@ -0,0 +1,37 @@
+'use strict';
+
+/**
+ * Fenwick Tree (Binary Indexed Tree)
+ */
+function FenwickTree(length) {
+  this._elements = new Array(length + 1);
+  for (var i = 0; i < this._elements.length ; i++)
+    this._elements[i] = 0;
+}
+
+/**
+ * Adds value to the array at specified index in O(log n)
+ */
+FenwickTree.prototype.adjust = function (index, value) {
+  for (; index < this._elements.length ; index += (index&-index))
```

Comment:
I disagree that the bit operation is trivial by just looking at it. The heap also represents a tree inside an array (but uses shifts (i.e. multiplication and divisions) instead), so it would be nice to have a line comment.

If you could explain the purpose and general idea in a few lines it would be better than the link, but if that's not the case feel free to add the link, it's better than nothing.